Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,44 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+        public const int FirstNameColumn = 1;
+        public const int LastNameColumn = 2;
+        public const int EmailColumn = 3;
+        public const int StreetAddressColumn = 4;
+        public const int CityColumn = 5;
+        public const int StateColumn = 6;
+        public const int ZipColumn = 7;
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows => File.ReadAllLines("People.csv").Skip(1);
+
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+            => CsvRows.Select(row => row.Split(",")[6]).OrderBy(State => State).Distinct();
 
         // 3.
         public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+            => string.Join(", ", GetUniqueSortedListOfStatesGivenCsvRows().ToArray());
 
         // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+        public IEnumerable<IPerson> People => CsvRows.OrderBy(streetAddress => streetAddress).Select(row => row.Split(","))
+            .OrderBy(state => state[StateColumn]).ThenBy(city => city[CityColumn]).ThenBy(zip => zip[ZipColumn])
+                .Select(person => new Person(person[FirstNameColumn], person[LastNameColumn],
+                    new Address(person[StreetAddressColumn], person[CityColumn], person[StateColumn], person[ZipColumn]), person[EmailColumn]));

```

Comment:
```suggestion
        public IEnumerable<IPerson> People 
            => CsvRows
            .OrderBy(streetAddress => streetAddress)
            .Select(row => row.Split(","))
            .OrderBy(state => state[StateColumn])
            .ThenBy(city => city[CityColumn])
            .ThenBy(zip => zip[ZipColumn])
                .Select(person => new Person(
                    person[FirstNameColumn], 
                    person[LastNameColumn],
                    new Address(
                        person[StreetAddressColumn], 
                        person[CityColumn], 
                        person[StateColumn], 
                        person[ZipColumn]
                    ), 
                    person[EmailColumn]
                ));
```
For improved readability consider splitting this across multiple lines.